geppetto.js/geppetto-ui/src/3d-canvas/showcase/examples/SelectionUtils.js

```diff
@@ -0,0 +1,40 @@
+import { hexToRGBNormalized } from "../../threeDEngine/util";
+
+const SELECTION_COLOR = { r: 0.8, g: 0.8, b: 0, a: 1 };
+
+
+export function dataMapping (data){
+  return data.map(item => (
+    {
+      color: item.selected ? SELECTION_COLOR : item.color ? item.color : hexToRGBNormalized(GEPPETTO.Resources.COLORS.DEFAULT),
+      instancePath: item.instancePath
+    }
+  ))
+}
+
+export function onSelection (selectedInstances) {
+  const { data } = this.state
+  const newData = []
+  let newInstance = true
+  for (const si of selectedInstances){
```

Comment:
we can write this more concise & avoid nested for-loop

by mapping `selectedInstances` to a Map
```
const map = new Map(selectedInstances.map(i => [i.instancePath, i.val]))
```

and iterating once over data and testing for `data.instancePath in map`.

Instead of newData & push you could also do
```
return data.map(item => {
})
```

For what do we use the `newInstance` flag here? 🤔

Reply:
I think your suggestion assumes that data is a superset of selectedInstances but as explained below, we can have selectedInstances that are not in data.
Either way, I modified this code a bit in a way that it 'hides' the nested for-loop and makes it more concise/readable. 
I think it's somewhat worse in terms of performance but it doesn't matter much for these small data structures.

Reply:
and to get the best of both worlds:

- one map for selected instances
- one map for data
- one for loop with the previous if-else and append logic but with O(1) lookup instead of nested-loop